### DISCUSSION TITLE
Accessing filename of a reference object

### DISCUSSION THREAD

**Discussion by haowens** [tinacms/tinacms#4019] [Help]
is there any reason why the auto-generated Post fragment won't let me access the _sys {filename} property on each category reference type in the categories?
<img width="369" alt="image" src="https://github.com/tinacms/tinacms/assets/69762131/f20b1017-aada-4fde-8994-b7caf8ade570">

the content api let's me query it:
<img width="1440" alt="image" src="https://github.com/tinacms/tinacms/assets/69762131/18d6fcde-3cb1-44c6-9764-0842d9fa58b6">

but when i try to write my own in line custom query that asks for that data the page doesn't render and it says that: something "cannot be serialized as JSON"

**Comment by jeffsee55** [comment]
Support for this has just been released!
